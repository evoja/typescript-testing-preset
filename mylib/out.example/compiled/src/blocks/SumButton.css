.evjttp-blocks-SumButton__red {
    color: red;
}

/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlN1bUJ1dHRvbi5tLmNzcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQTtJQUNJLFVBQVU7QUFDZCIsImZpbGUiOiJTdW1CdXR0b24ubS5jc3MiLCJzb3VyY2VzQ29udGVudCI6WyIucmVkIHtcbiAgICBjb2xvcjogcmVkO1xufVxuIl19 */