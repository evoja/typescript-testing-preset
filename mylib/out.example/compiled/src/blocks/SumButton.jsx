"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const sum_1 = require("@evoja/typescript-testing-preset--lib/tools/sum");
const SumButton_m_css_1 = __importDefault(require("./SumButton.m.css"));
function SumButton(props) {
    const { a, b, red } = props;
    return (<button className={red ? SumButton_m_css_1.default['red'] : ''}>
      {`sum(${a}, ${b}) -> ${sum_1.sum(a, b)}`}
    </button>);
}
exports.default = SumButton;
//# sourceMappingURL=SumButton.jsx.map