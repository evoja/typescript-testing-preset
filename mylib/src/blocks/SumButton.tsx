import {sum} from '@evoja/typescript-testing-preset--lib/tools/sum'
import cls from './SumButton.m.css'

type SumButtonProps = {
  a: number
  b: number
  red?: boolean
}

export default function SumButton(props: SumButtonProps) {
  const {a, b, red} = props
  return (
    <button className={red ? cls['red'] : ''}>
      {`sum(${a}, ${b}) -> ${sum(a,b)}`}
    </button>
  )
}
