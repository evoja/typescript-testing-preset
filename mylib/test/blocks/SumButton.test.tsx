/**
 * @jest-environment jsdom
 */

import { act } from 'react-dom/test-utils'
import {create} from 'react-test-renderer'
import { toMatchImageSnapshot } from 'jest-image-snapshot'
expect.extend({ toMatchImageSnapshot });
import {render} from 'react-dom'
import SumButton from '@evoja/typescript-testing-preset--lib/blocks/SumButton'

test('button', () => {
  const container = document.createElement('div')
  document.body.appendChild(container)
  const button_elem = <SumButton a={10} b={20}/>
  act(() => {render(button_elem, container)});
  const button = container.querySelector('button');
  expect(button?.textContent).toBe('sum(10, 20) -> 30');
  expect(button?.textContent).not.toBe('sum(10, 20) -> 40');

  expect(button?.outerHTML).toMatchSnapshot();
  const rendered = create(button_elem);
  expect(rendered).toMatchSnapshot();
  expect(rendered.toJSON()).toMatchSnapshot();
  expect(rendered.toTree()).toMatchSnapshot();

  document.body.removeChild(container)
})
