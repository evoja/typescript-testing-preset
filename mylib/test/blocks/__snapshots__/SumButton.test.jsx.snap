// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button 1`] = `"<button class=\\"\\">sum(10, 20) -&gt; 30</button>"`;

exports[`button 2`] = `
<button
  className=""
>
  sum(10, 20) -&gt; 30
</button>
`;

exports[`button 3`] = `
<button
  className=""
>
  sum(10, 20) -&gt; 30
</button>
`;

exports[`button 4`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "a": 10,
    "b": 20,
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": "sum(10, 20) -> 30",
      "className": "",
    },
    "rendered": Array [
      "sum(10, 20) -> 30",
    ],
    "type": "button",
  },
  "type": [Function],
}
`;
